src/scene_battle_rpg2k.cpp

```diff
@@ -450,12 +461,12 @@ bool Scene_Battle_Rpg2k::ProcessBattleAction(Game_BattleAlgorithm::AlgorithmBase
 			}
 
 			if (action->TargetNext()) {
-				battle_action_state = BattleActionState_Start;
+				battle_action_state = BattleActionState_ConditionHeal;
 				return false;
 			}
 
 			// Reset variables
-			battle_action_state = BattleActionState_Start;
+			battle_action_state = BattleActionState_ConditionHeal;
```

Comment:
In my tests your state changes broke the message box. At the beginning it shows an empty message box with a delay and between battle actions, too.

And after an enemy finished his battle animation the delay got removed.